core/trino-main/src/test/java/io/trino/execution/TestCreateCatalogTask.java

```diff
@@ -0,0 +1,152 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.execution.warnings.WarningCollector;
+import io.trino.plugin.tpch.TpchConnectorFactory;
+import io.trino.security.AllowAllAccessControl;
+import io.trino.spi.TrinoException;
+import io.trino.spi.connector.Connector;
+import io.trino.spi.connector.ConnectorContext;
+import io.trino.spi.connector.ConnectorFactory;
+import io.trino.spi.resourcegroups.ResourceGroupId;
+import io.trino.sql.tree.CatalogProperty;
+import io.trino.sql.tree.CreateCatalog;
+import io.trino.sql.tree.QualifiedName;
+import io.trino.sql.tree.StringLiteral;
+import io.trino.testing.LocalQueryRunner;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.util.concurrent.MoreExecutors.directExecutor;
+import static io.airlift.concurrent.MoreFutures.getFutureValue;
+import static io.trino.SessionTestUtils.TEST_SESSION;
+import static java.util.Collections.emptyList;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.testng.Assert.assertTrue;
+
+@Test(singleThreaded = true)
+public class TestCreateCatalogTask
+{
+    private static final String TEST_CATALOG = "test_catalog";
+    private static final ImmutableList<CatalogProperty> TPCH_PROPERTIES = ImmutableList.of(new CatalogProperty("connector.name", new StringLiteral("tpch")));
+
+    protected LocalQueryRunner queryRunner;
+    private QueryStateMachine queryStateMachine;
+
+    @BeforeMethod
+    public void setUp()
+    {
+        queryRunner = LocalQueryRunner.create(TEST_SESSION);
+        queryRunner.registerCatalogFactory(new TpchConnectorFactory());
+        queryRunner.registerCatalogFactory(new FailConnectorFactory());
+        queryStateMachine = QueryStateMachine.begin(
+                Optional.empty(),
+                "test",
+                Optional.empty(),
+                queryRunner.getDefaultSession(),
+                URI.create("fake://uri"),
+                new ResourceGroupId("test"),
+                false,
+                queryRunner.getTransactionManager(),
+                queryRunner.getAccessControl(),
+                directExecutor(),
+                queryRunner.getMetadata(),
+                WarningCollector.NOOP,
+                Optional.empty());
+    }
+
+    @AfterMethod
+    public void tearDown()
+    {
+        if (queryRunner != null) {
+            queryRunner.close();
+        }
```

Comment:
I was interested in this PR and ran the test, and the test failed with the error below.

```
2022-12-06T16:57:34.501+0900	INFO	main	stderr	FATAL: io.trino.testng.services.ManageTestResources: 
	Test instance field has value that looks like a resource
	    Test class: class io.trino.execution.TestCreateCatalogTask
	    Instance: io.trino.execution.TestCreateCatalogTask@7f0ebc40
	    Field: protected io.trino.testing.LocalQueryRunner io.trino.execution.TestCreateCatalogTask.queryRunner
	    Value: io.trino.testing.LocalQueryRunner@17ac3bf1
	    Test execution stage: AFTER_CLASS
	    Matching rule: isAutoCloseable

	Resources must not be allocated in a field initializer or a test constructor,
	and must be freed when test class is completed.
	Depending which rule has been violated, if the reported class holds on
	to resources only conditionally, you can add @ResourcePresence to declare that.


2022-12-06T16:57:34.501+0900	INFO	main	stderr	JVM will be terminated
```

It looks like the code below is missing.
```
 queryRunner = null;
 queryStateMachine = null;
```

Same issue with TestDropCatalogTask.

I hope this helps.